src/test/spec/unified_runner/test_file.rs

```diff
@@ -0,0 +1,254 @@
+#![allow(dead_code)]
+
+use semver::{Version, VersionReq};
+use serde::{Deserialize, Deserializer};
+
+use super::TestEvent;
+
+use crate::{
+    bson::{Bson, Document},
+    error::Error,
+    options::{
+        ClientOptions,
+        CollectionOptions,
+        DatabaseOptions,
+        ReadConcern,
+        ReadPreference,
+        SelectionCriteria,
+        WriteConcern,
+    },
+    test::TestClient,
+};
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TestFile {
+    pub description: String,
+    #[serde(deserialize_with = "deserialize_schema_version")]
+    pub schema_version: Version,
+    pub run_on_requirements: Option<Vec<RunOnRequirement>>,
+    pub allow_multiple_mongoses: Option<bool>,
+    pub create_entities: Option<Vec<Entity>>,
+    pub initial_data: Option<Vec<CollectionData>>,
+    pub tests: Vec<TestCase>,
+}
+
+fn deserialize_schema_version<'de, D>(deserializer: D) -> std::result::Result<Version, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let mut schema_version = String::deserialize(deserializer)?;
+    // if the schema version only contains a major and minor version (e.g. 1.0), append a ".0" to
+    // ensure correct parsing into a semver::Version
+    if schema_version.split('.').count() == 2 {
+        schema_version.push_str(".0");
+    }
+    Version::parse(&schema_version).map_err(|e| serde::de::Error::custom(format!("{}", e)))
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RunOnRequirement {
+    pub min_server_version: Option<String>,
```

Comment:
do these fields need to be `pub`? If this type will be exclusively accessed via `can_run_on`, then the `pub` visibility can be removed.